python/pytests/execution_test.py

```diff
@@ -139,3 +140,33 @@ async def test_history(golden, source_int64) -> None:
             )
         )
     )
+
+async def test_iter_pandas_async_live_timeout(golden, source_int64) -> None:
+    data2 = "\n".join(
+        [
+            "time,key,m,n",
+            "1996-12-20T16:39:57,A,5,10",
+            "1996-12-20T16:39:58,B,24,3",
+            "1996-12-20T16:39:59,A,17,6",
+            "1996-12-20T16:40:00,C,,9",
+            "1996-12-20T16:40:01,A,12,",
+            "1996-12-20T16:40:02,A,,",
+        ]
+    )
+
+    execution = source_int64.run_iter(mode="live")
+
+    # Await the first batch.
+    golden.jsonl(await execution.__anext__())
+
+    # Add data and await the second batch.
+    await source_int64.add_string(data2)
+    golden.jsonl(await execution.__anext__())
+
+    with pytest.raises(TimeoutError):
+        async with asyncio.timeout(1):
```

Comment:
I don't know if we want to check this in without marking it as a slow test. Specifically, it will wait at least 1 second for the timeout to occur. This can be increased further (10s) to verify that it isn't just "getting lucky" in terms of hitting an empty batch or somethnig like that.